stopBeforeObstacle/stopBeforeObstacle.ino

```diff
@@ -0,0 +1,49 @@
+#include <Smartcar.h>
+
+const int FRONT_TRIGGER_PIN = 18; 
+const int FRONT_ECHO_PIN = 5; 
+const unsigned int FRONT_MAX_DISTANCE = 100;
+SR04 front_sensor(FRONT_TRIGGER_PIN, FRONT_ECHO_PIN, FRONT_MAX_DISTANCE);
+
+BrushedMotor leftMotor(smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+
+const auto pulsesPerMeter = 600;
+
+DirectionlessOdometer leftOdometer(
+    smartcarlib::pins::v2::leftOdometerPin, []() { leftOdometer.update(); }, pulsesPerMeter);
+DirectionlessOdometer rightOdometer(
+    smartcarlib::pins::v2::rightOdometerPin, []() { rightOdometer.update(); }, pulsesPerMeter);
+
+DistanceCar car(control, leftOdometer, rightOdometer);
+
+void setup()
+{
+    Serial.begin(9600);
+    car.enableCruiseControl();
+}
+
+void loop()
+{
+    car.update();
+    checkObstacle();
+}
+
+void checkObstacle(){
+
+    //We set the speed to 1 m/s when the distance to the obstacle is < 25cm
+    //OR the distance is 0, we do this because of the sensor "bug"
+    //Where the distance is < 1m it outputs 0.
+    //Else we stop the car.
+    
+    if(front_sensor.getDistance() > 25 || front_sensor.getDistance() == 0){
+    
```

Comment:
This will make two measurements which will not be necessarily the same.
What if the first call to getDistance returns 0 (e.g there is no obstacle nearby) and the second one 100?

Reply:
I'm not 100% sure I understand, do you mean that in the If loop we will get two instances/measurements of the distance? We tested the code and it worked the way we expected.

I guess we could create a variable, int distance = front_sensor.getDistance();, and then place that variable in the two conditions.

Reply:
> I guess we could create a variable, int distance = front_sensor.getDistance();, and then place that variable in the two conditions.

Exactly. 

> We tested the code and it worked the way we expected.

The fact that it happened to work, doesn't mean it always would. :)
This is why you should be writing many and short automated tests, instead of manually trying things out. :+1:

Reply:
Alright! So, what happens now? Should we remove the pull request and do change the code and then merge again before the end of the day? Or can we agree to the changes and change them if/when we need it?

Reply:
Ahhh, it looks like I can change the file here in GitHub, is that the next move?

Reply:
> Ahhh, it looks like I can change the file here in GitHub, is that the next move?

You can do that since it's a small change but generally you would just commit again on the same branch and push. Then thus pull request would be updated.